inc/EventDrivenPrognoser.h

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2018 United States Government as represented by the
+// Administrator of the National Aeronautics and Space Administration.
+// All Rights Reserved.
+#ifndef PCOE_EVENTDRIVENPROGNOSER_H
+#define PCOE_EVENTDRIVENPROGNOSER_H
+#include <memory>
+
+#include "GSAPConfigMap.h"
+#include "LoadEstimatorFactory.h"
+#include "Messages/MessageBus.h"
+#include "Observers/EventDrivenObserver.h"
+#include "Observers/ObserverFactory.h"
+#include "Predictors/EventDrivenPredictor.h"
+#include "Predictors/PredictorFactory.h"
+#include "PrognosticsModelFactory.h"
+
+namespace PCOE {
+    const std::string SOURCE_KEY = "Source";
+    const std::string MODEL_KEY = "Model";
+    const std::string OBSERVER_KEY = "Observer";
+    const std::string PREDICTOR_KEY = "Predictor";
+    const std::string LOAD_EST_KEY = "LoadEstimator";
+
+    /**
+     * Manages event driven components.
+     *
+     * @author Jason Watkins
+     * @since 1.2
+     **/
+    class EventDrivenPrognoser {
+    public:
+        // TODO (JW): Figure out how to make this sane
+        EventDrivenPrognoser(MessageBus& bus, const GSAPConfigMap& config)
+            : bus(bus),
+              model(PrognosticsModelFactory::instance().Create(config.at(MODEL_KEY)[0], config)),
+              loadEstimator(
```

Comment:
Why doesn't the load estimator get the bus as an argument? Isn't it going to receive data from the bus?

Reply:
I see that this isn't implemented in this version (and is not needed for the existing load estimators. Is this planned in the future?

Reply:
Yes, this change will have to be made when the flight-plan based load estimator is added.